Add SPI for WTF::userVisibleString in WebKit.framework

achristensen07 opened the pull request:
#### 0d2a7ef45b96b34fbd77c9bfeceb53f266a15405
<pre>
Add SPI for WTF::userVisibleString in WebKit.framework
<a href="https://bugs.webkit.org/show_bug.cgi?id=290884">https://bugs.webkit.org/show_bug.cgi?id=290884</a>

Reviewed by Richard Robinson.

This is a step towards being able to remove WebKitLegacy on macOS Recovery
This SPI does the same thing as URL.FormatStyle but it is updated with Safari.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(+[WKWebView _userVisibleStringForURL:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WTF/cocoa/URLExtras.mm:
(TestWebKitAPI::userVisibleString):
(TestWebKitAPI::TEST(URLExtras, URLExtras)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_Spoof)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_NotSpoofed)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_DivisionSign)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_Space)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_File)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_ParsingError)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_Nil)):
(TestWebKitAPI::TEST(URLExtras, CreateNSArray)):
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_Spoof)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_NotSpoofed)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_DivisionSign)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_Space)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_File)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_ParsingError)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_Nil)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, CreateNSArray)): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ScreenTime.mm:

Canonical link: <a href="https://commits.webkit.org/293417@main">https://commits.webkit.org/293417@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bd295b18823c5d50cbb9f69cc8b506addd3a9736

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/98846 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/18483 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8724 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/103972 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/49435 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/100891 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18778 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/26933 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75251 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32385 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/101850 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14267 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89267 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55611 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14058 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7239 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/48815 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/83997 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7310 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106341 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/25943 "Built successfully") | [⏳ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/macOS-Sequoia-Debug-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84216 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26318 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85465 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83714 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28364 "Passed tests") | [⏳ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/macOS-Sonoma-Release-WK2-Intel-Tests-EWS "Waiting to run tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/19657 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/16069 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/25896 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/31082 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25716 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29036 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27290 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
```diff
@@ -436,6 +436,8 @@ for this property.
 - (void)_convertPoint:(CGPoint)point fromFrame:(WKFrameInfo *)frame toMainFrameCoordinates:(void (^)(CGPoint, NSError *error))completionHandler WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA));
 - (void)_convertRect:(CGRect)rect fromFrame:(WKFrameInfo *)frame toMainFrameCoordinates:(void (^)(CGRect, NSError *error))completionHandler WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA));
 
++ (NSString *)_userVisibleStringForURL:(NSURL *)url WK_API_AVAILABLE(macos(WK_MAC_TBA), ios(WK_IOS_TBA), visionos(WK_XROS_TBA));
```
can this ever return nil? if so should it be marked nullable?

achristensen07 replied:
If I added nullability annotations for this method, I'd have to add them for all methods in this header.

rr-codes replied:
maybe we could add NS_ASSUME_NONNULL_BEGIN / NS_ASSUME_NONNULL_END to this file and then it would be fine to only add the nullability annotation to this method?

achristensen07 replied:
That would make it start assuming that everything without annotation is non-null, which is certainly incorrect.  I would have to go through all the selectors and annotate them, which is why it hasn't been done yet.

achristensen07 committed:
Add SPI for WTF::userVisibleString in WebKit.framework
https://bugs.webkit.org/show_bug.cgi?id=290884

Reviewed by Richard Robinson.

This is a step towards being able to remove WebKitLegacy on macOS Recovery
This SPI does the same thing as URL.FormatStyle but it is updated with Safari.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(+[WKWebView _userVisibleStringForURL:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewPrivate.h:
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WTF/cocoa/URLExtras.mm:
(TestWebKitAPI::userVisibleString):
(TestWebKitAPI::TEST(URLExtras, URLExtras)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_Spoof)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_NotSpoofed)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_DivisionSign)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_Space)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_File)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_ParsingError)):
(TestWebKitAPI::TEST(URLExtras, URLExtras_Nil)):
(TestWebKitAPI::TEST(URLExtras, CreateNSArray)):
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_Spoof)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_NotSpoofed)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_DivisionSign)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_Space)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_File)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_ParsingError)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, URLExtras_Nil)): Deleted.
(TestWebKitAPI::TEST(WTF_URLExtras, CreateNSArray)): Deleted.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/ScreenTime.mm:

Canonical link: https://commits.webkit.org/293417@main
